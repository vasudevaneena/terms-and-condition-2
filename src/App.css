* {
    border: 0;
    text-decoration: none;
    text-size-adjust: 100%;
    -webkit-tap-highlight-color: transparent;
    padding: 0;
    margin: 0;
    outline: none;
}

html,
button {
    font-size: 16px;
    line-height: 24px;
    font-family: -apple-system, BlinkMacSystemFont, "Helvetica Neue", Helvetica, sans-serif;
    color: #21184B;
    -webkit-font-smoothing: subpixel-antialiased;
    appearance: none;
    touch-action: manipulation;
}

header {
    background-color: rgba(244, 243, 246, 0.95);
    border-bottom: 2px solid #E9E8ED;
    text-align: center;
    padding: 12px;
}

h1 {
    font-size: 18px;
}

form {
    display: flex;
    flex-direction: column;
    max-width: 600px;
    margin: 0 auto;
}

label {
    position: relative;
    padding: 0 12px 0 30px;
    margin: 12px 24px;
}


.input-style {
    background: #FFFFFF;
    border: 2px solid rgba(33, 24, 75, 0.2);
    border-radius: 3px;
    box-sizing: border-box;
    padding: 12px 12px 12px 28px;
    margin: 6px -12px 0 -30px;
    width: calc(100% + 42px);
}
.check-padding{
    padding: 12px 12px 12px 28px;
}

input::placeholder {
    color: rgba(176, 167, 218, 0.2);
}

input:focus {
    border-color: #22A8AC;
    box-shadow: 0 0 0 3px rgba(34, 168, 172, 0.2);
}

.cta {
    display: block;
    border-radius: 3px;
    background-color: #22A8AC;
    font-weight: bold;
    color: #FFFFFF;
    text-align: center;
    white-space: nowrap;
    text-overflow: ellipsis;
    overflow: hidden;
    padding: 12px 30px;
    margin: 12px 24px;
    cursor: pointer;
}

.disable-cta{
    padding: 12px 30px;
    margin: 12px 24px;
}

.cta:hover {
    background-color: #1D8D90;
}
