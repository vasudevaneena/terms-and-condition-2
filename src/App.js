import React, { Component } from "react";
import "./App.css";

class App extends Component {
  constructor(props) {
    super(props);
    this.user = {
      email: "",
      password: ""
    };
    this.state = {
      disable: true,
      checkbox: false,
      classname: "disable-cta"
    };

    this.handleChecked = this.handleChecked.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.updateEmail = this.updateField.bind(this, "email");
  }
  updateField(name, event) {
    const value = event.currentTarget.value;
    this.user[name] = value;
  }
  registerUser(user) {
    // Mocked response of user submission to register
    return new Promise(resolve => {
      let response = "User has been successfully registered";
      const errors = { email: [], password: [] };
      const validEmail = /\S+@\S+\.\S+/.test(user.email);

      if (!user.email) {
        errors.email.push("Email required");
      } else if (!validEmail) {
        errors.email.push("Email must be valid");
      }

      if (!user.password) {
        errors.password.push("Password required");
      } else if (user.password.length < 8) {
        errors.password.push("Password must contain at least 8 characters");
      }

      for (const errorType in errors) {
        if (errors[errorType].length) {
          response = errors;
          break;
        }
      }

      setTimeout(() => {
        return resolve(response);
      }, Math.random() * 1000 + 500);
    });
  }

  async handleSubmit(event) {
    event.preventDefault();
    const response = await this.registerUser(this.user);
  }
  handleChecked(event) {
    const isChecked = event.target.checked;
    if (isChecked) {
      this.setState({ disable: false, checkbox: true, classname: "cta" });
      //document.getElementById("registerButton").className = "cta";
    } else {
      this.setState({
        disable: true,
        checkbox: false,
        classname: "disable-cta"
      });
      //document.getElementById("registerButton").className = "disable-cta";
    }
  }

  render() {
    return (
      <section>
        <header>
          <h1>HealthShare Front-End Test</h1>
        </header>
        <form onSubmit={this.handleSubmit}>
          <label>
            Email address
            <input
              className="input-style"
              name="email"
              onChange={this.updateEmail}
              placeholder="e.g name@example.com"
            />
          </label>
          <label>
            Password
            <input
              className="input-style"
              name="password"
              placeholder="••••••••"
            />
          </label>
          <div className="check-padding">
            <input
              type="checkbox"
              name="checkbox"
              id="checkbox"
              value={this.state.checkbox}
              onChange={this.handleChecked}
            />
            <a> Accept terms and condition to Register</a>
          </div>
          <button
            type="submit"
            id="registerButton"
            className={this.state.classname}
            disabled={this.state.disable}
          >
            Register
          </button>
        </form>
      </section>
    );
  }
}

export default App;
