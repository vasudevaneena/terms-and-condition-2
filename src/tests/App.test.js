import React from "react";
import Adapter from "enzyme-adapter-react-16";
import { configure, shallow, mount } from "enzyme";
import App from "../App";

configure({ adapter: new Adapter() });

let wrapper;

beforeEach(() => {
  wrapper = shallow(<App />);
});

describe("App snapshot", () => {
  test("matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe("<App /> rendering", () => {
  it("should render one <section>", () => {
    expect(wrapper.find("section")).toHaveLength(1);
  });
  it("should render one <header>", () => {
    expect(wrapper.find("header")).toHaveLength(1);
  });

  it("should render one <form>", () => {
    expect(wrapper.find("form")).toHaveLength(1);
  });
  it("should render 3 <input>s", () => {
    expect(wrapper.find("input")).toHaveLength(3);
  });
  it("should render 1 <anchor>", () => {
    expect(wrapper.find("a")).toHaveLength(1);
  });
  it("should render 2 <label>s", () => {
    expect(wrapper.find("label")).toHaveLength(2);
  });
  it("should render 1 <button>", () => {
    expect(wrapper.find("button")).toHaveLength(1);
  });
  it("should render 1 <checkbox>s", () => {
    expect(wrapper.find('input[type="checkbox"]')).toHaveLength(1);
  });
});

describe("Form state", function() {
  it(" button is disabled initially", () => {
    expect(wrapper.state("disable")).toEqual(true);
  });
  it(" checkbox uncehcked initially", () => {
    expect(wrapper.state("checkbox")).toEqual(false);
  });

  it("click checkbox", () => {
    wrapper = shallow(<App />, { attachTo: document.body });
    wrapper.find("#checkbox").simulate("change", { target: { checked: true } });
    expect(wrapper.state("checkbox")).toEqual(true);
    expect(wrapper.state("disable")).toEqual(false);
  });
});
