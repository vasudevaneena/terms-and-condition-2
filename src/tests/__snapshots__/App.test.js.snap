// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header>
          <h1>
            HealthShare Front-End Test
          </h1>
        </header>,
        <form
          onSubmit={[Function]}
        >
          <label>
            Email address
            <input
              className="input-style"
              name="email"
              onChange={[Function]}
              placeholder="e.g name@example.com"
            />
          </label>
          <label>
            Password
            <input
              className="input-style"
              name="password"
              placeholder="••••••••"
            />
          </label>
          <div
            className="check-padding"
          >
            <input
              id="checkbox"
              name="checkbox"
              onChange={[Function]}
              type="checkbox"
              value={false}
            />
            <a>
               Accept terms and condition to Register
            </a>
          </div>
          <button
            className="disable-cta"
            disabled={true}
            id="registerButton"
            type="submit"
          >
            Register
          </button>
        </form>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1>
            HealthShare Front-End Test
          </h1>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "HealthShare Front-End Test",
          },
          "ref": null,
          "rendered": "HealthShare Front-End Test",
          "type": "h1",
        },
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label>
              Email address
              <input
                className="input-style"
                name="email"
                onChange={[Function]}
                placeholder="e.g name@example.com"
              />
            </label>,
            <label>
              Password
              <input
                className="input-style"
                name="password"
                placeholder="••••••••"
              />
            </label>,
            <div
              className="check-padding"
            >
              <input
                id="checkbox"
                name="checkbox"
                onChange={[Function]}
                type="checkbox"
                value={false}
              />
              <a>
                 Accept terms and condition to Register
              </a>
            </div>,
            <button
              className="disable-cta"
              disabled={true}
              id="registerButton"
              type="submit"
            >
              Register
            </button>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Email address",
                <input
                  className="input-style"
                  name="email"
                  onChange={[Function]}
                  placeholder="e.g name@example.com"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Email address",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "input-style",
                  "name": "email",
                  "onChange": [Function],
                  "placeholder": "e.g name@example.com",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Password",
                <input
                  className="input-style"
                  name="password"
                  placeholder="••••••••"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Password",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "input-style",
                  "name": "password",
                  "placeholder": "••••••••",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  id="checkbox"
                  name="checkbox"
                  onChange={[Function]}
                  type="checkbox"
                  value={false}
                />,
                <a>
                   Accept terms and condition to Register
                </a>,
              ],
              "className": "check-padding",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "id": "checkbox",
                  "name": "checkbox",
                  "onChange": [Function],
                  "type": "checkbox",
                  "value": false,
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": " Accept terms and condition to Register",
                },
                "ref": null,
                "rendered": " Accept terms and condition to Register",
                "type": "a",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Register",
              "className": "disable-cta",
              "disabled": true,
              "id": "registerButton",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Register",
            "type": "button",
          },
        ],
        "type": "form",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header>
            <h1>
              HealthShare Front-End Test
            </h1>
          </header>,
          <form
            onSubmit={[Function]}
          >
            <label>
              Email address
              <input
                className="input-style"
                name="email"
                onChange={[Function]}
                placeholder="e.g name@example.com"
              />
            </label>
            <label>
              Password
              <input
                className="input-style"
                name="password"
                placeholder="••••••••"
              />
            </label>
            <div
              className="check-padding"
            >
              <input
                id="checkbox"
                name="checkbox"
                onChange={[Function]}
                type="checkbox"
                value={false}
              />
              <a>
                 Accept terms and condition to Register
              </a>
            </div>
            <button
              className="disable-cta"
              disabled={true}
              id="registerButton"
              type="submit"
            >
              Register
            </button>
          </form>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1>
              HealthShare Front-End Test
            </h1>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "HealthShare Front-End Test",
            },
            "ref": null,
            "rendered": "HealthShare Front-End Test",
            "type": "h1",
          },
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label>
                Email address
                <input
                  className="input-style"
                  name="email"
                  onChange={[Function]}
                  placeholder="e.g name@example.com"
                />
              </label>,
              <label>
                Password
                <input
                  className="input-style"
                  name="password"
                  placeholder="••••••••"
                />
              </label>,
              <div
                className="check-padding"
              >
                <input
                  id="checkbox"
                  name="checkbox"
                  onChange={[Function]}
                  type="checkbox"
                  value={false}
                />
                <a>
                   Accept terms and condition to Register
                </a>
              </div>,
              <button
                className="disable-cta"
                disabled={true}
                id="registerButton"
                type="submit"
              >
                Register
              </button>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Email address",
                  <input
                    className="input-style"
                    name="email"
                    onChange={[Function]}
                    placeholder="e.g name@example.com"
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Email address",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "input-style",
                    "name": "email",
                    "onChange": [Function],
                    "placeholder": "e.g name@example.com",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Password",
                  <input
                    className="input-style"
                    name="password"
                    placeholder="••••••••"
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Password",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "input-style",
                    "name": "password",
                    "placeholder": "••••••••",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    id="checkbox"
                    name="checkbox"
                    onChange={[Function]}
                    type="checkbox"
                    value={false}
                  />,
                  <a>
                     Accept terms and condition to Register
                  </a>,
                ],
                "className": "check-padding",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "id": "checkbox",
                    "name": "checkbox",
                    "onChange": [Function],
                    "type": "checkbox",
                    "value": false,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " Accept terms and condition to Register",
                  },
                  "ref": null,
                  "rendered": " Accept terms and condition to Register",
                  "type": "a",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Register",
                "className": "disable-cta",
                "disabled": true,
                "id": "registerButton",
                "type": "submit",
              },
              "ref": null,
              "rendered": "Register",
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
